
syntax = "proto3";

package Lya.protos;

service LyaService {
    rpc extract(PBSyncRequest) returns (PBSyncResponse);
    rpc compile(PBCompileRequest) returns (PBCompileResponse);
    rpc check_availability(PBAvailabilityRequest) returns (PBAvailabilityResponse);
}

message PBAvailabilityRequest { }

message PBAvailabilityResponse { }

message PBSyncRequest {
    repeated string functions = 1;
    repeated string files = 2;
    uint64 start_line = 3;
}

message PBCompileRequest {
    repeated string messages = 1;
}

message PBCompileResponse { }

message PBSyncResponse {
    repeated PBFileToLocalizations file_to_localizations = 1;
    repeated PBDiagnostic diagnostics = 2;
}

message PBFileToLocalizations {
    string file = 1;
    repeated PBLocalization localizations = 2;
}

message PBLocalization {
    string id = 1;
    string text = 2;
    repeated PBParam parameters = 3;
    PBLocation location = 4;
}

message PBParam {
    string name = 1;
    bool is_list = 2;
    string type = 3;
}

message PBDiagnostic {
    string message = 1;
    PBLocation location = 2;
}

message PBLocation {
    uint64 line = 1;
    uint64 column = 2;
    uint64 length = 3;
    uint64 position = 4;
}